pkg/api/scylla/validation/scylladbcluster_validation.go

```diff
@@ -0,0 +1,450 @@
+		if dcTemplate.ScyllaDB.Storage != nil {
+			if dcTemplate.ScyllaDB.Storage.Metadata != nil {
+				allErrs = append(allErrs, apimachinerymetav1validation.ValidateLabels(dcTemplate.ScyllaDB.Storage.Metadata.Labels, fldPath.Child("scyllaDB", "storage", "metadata", "labels"))...)
+				allErrs = append(allErrs, apimachineryvalidation.ValidateAnnotations(dcTemplate.ScyllaDB.Storage.Metadata.Annotations, fldPath.Child("scyllaDB", "storage", "metadata", "annotations"))...)
+			}
+
+			storageCapacity, err := resource.ParseQuantity(dcTemplate.ScyllaDB.Storage.Capacity)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("scyllaDB", "storage", "capacity"), dcTemplate.ScyllaDB.Storage.Capacity, fmt.Sprintf("unable to parse capacity: %v", err)))
+			} else if storageCapacity.CmpInt64(0) <= 0 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("scyllaDB", "storage", "capacity"), dcTemplate.ScyllaDB.Storage.Capacity, "must be greater than zero"))
+			}
+
+			if dcTemplate.ScyllaDB.Storage.StorageClassName != nil {
+				for _, msg := range apimachineryvalidation.NameIsDNSSubdomain(*dcTemplate.ScyllaDB.Storage.StorageClassName, false) {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("scyllaDB", "storage", "storageClassName"), *dcTemplate.ScyllaDB.Storage.StorageClassName, msg))
+				}
+			}
+		}
+
+		if dcTemplate.ScyllaDB.CustomConfigMapRef != nil {
+			for _, msg := range apimachineryvalidation.NameIsDNSSubdomain(*dcTemplate.ScyllaDB.CustomConfigMapRef, false) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("scyllaDB", "customConfigMapRef"), *dcTemplate.ScyllaDB.CustomConfigMapRef, msg))
+			}
+		}
+	}
+
+	if dcTemplate.ScyllaDBManagerAgent != nil {
+		if dcTemplate.ScyllaDBManagerAgent.CustomConfigSecretRef != nil {
+			for _, msg := range apimachineryvalidation.NameIsDNSSubdomain(*dcTemplate.ScyllaDBManagerAgent.CustomConfigSecretRef, false) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("scyllaDBManagerAgent", "customConfigSecretRef"), *dcTemplate.ScyllaDBManagerAgent.CustomConfigSecretRef, msg))
+			}
+		}
+	}
+
+	return allErrs
+}
+
+func ValidateScyllaDBClusterSpecExposeOptions(options *scyllav1alpha1.ScyllaDBClusterExposeOptions, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if options.NodeService != nil {
+		allErrs = append(allErrs, ValidateScyllaDBClusterNodeService(options.NodeService, fldPath)...)
+	}
+
+	if options.BroadcastOptions != nil {
+		allErrs = append(allErrs, ValidateScyllaDBClusterNodeBroadcastOptions(options.BroadcastOptions, options.NodeService, fldPath.Child("broadcastOptions"))...)
+	}
+
+	return allErrs
+}
+
+func ValidateScyllaDBClusterNodeService(nodeService *scyllav1alpha1.NodeServiceTemplate, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	var supportedServiceTypes = []scyllav1alpha1.NodeServiceType{
+		scyllav1alpha1.NodeServiceTypeHeadless,
+		scyllav1alpha1.NodeServiceTypeLoadBalancer,
+	}
```

Comment:
clusterIP should also be supported: https://github.com/scylladb/scylla-operator-enterprise/pull/41#discussion_r1688275789